instrumentation/testing/testing.go

```diff
@@ -168,3 +173,79 @@ func (test *Test) Run(name string, f func(t *testing.T)) {
 		f(childT)
 	})
 }
+
+//Extract benchmark result from the private result field in testing.B
+func extractBenchmarkResult(b *testing.B) (*testing.BenchmarkResult, error) {
+	val := reflect.Indirect(reflect.ValueOf(b))
+	member := val.FieldByName("result")
+	if member.IsValid() {
+		ptrToY := unsafe.Pointer(member.UnsafeAddr())
+		return (*testing.BenchmarkResult)(ptrToY), nil
+	}
+	return nil, stdErrors.New("result can't be retrieved")
+}
+
+// Starts a new benchmark using a pc s a caller
+func StartBenchmark(b *testing.B, pc uintptr, benchFunc func(b *testing.B)) {
+	var bChild *testing.B
+	b.ResetTimer()
+	b.ReportAllocs()
+	startTime := time.Now()
+	result := b.Run("*", func(b1 *testing.B) {
```

Comment:
I think is the cleanest way because the behavior of the Benchmark, calling the benchmark under test several times causing multiple `defer` calls. The only way I see is to check in each `defer` if we have results for that test or not yet (because we don't have any `done` flag on the `testing.B` struct)